core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -1444,13 +1483,136 @@ fn parse_str_to_time_only_timestamp(value: &str) -> CometResult<Option<i64>> {
     Ok(Some(timestamp))
 }
 
+//a string to date parser - port of spark's SparkDateTimeUtils#stringToDate.
+fn date_parser(date_str: &str, eval_mode: EvalMode) -> CometResult<Option<i32>> {
+    // local functions
+    fn get_trimmed_start(bytes: &[u8]) -> usize {
```

Comment:
Is this defined because we cannot use `String#trim()` or `trim_matches`?

Reply:
leave as is for now based on above comment from @andygrove